url_serde/src/lib.rs

```diff
@@ -179,8 +178,8 @@ impl<T> De<T> where De<T>: serde::Deserialize {
 }
 
 /// Deserializes this URL from a `serde` stream.
-impl Deserialize for De<Url> {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer {
+impl<'de> Deserialize<'de> for De<Url> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
         let string_representation: String = Deserialize::deserialize(deserializer)?;
```

Comment:
With serde 1.0 we might be able to use `Cow<'de, str>` here instead of `String` (https://serde.rs/borrow.html) and make one less memory allocation in most cases. But that can be done later.